File: liqp-core/src/test/java/liqp/filters/AbsTest.java
package liqp.filters;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

import liqp.Template;
import liqp.LiquidParser;
import org.antlr.runtime.RecognitionException;
import org.junit.Test;

public class AbsTest {

    /*
        def test_abs
          assert_template_result "17", "{{ 17 | abs }}"
          assert_template_result "17", "{{ -17 | abs }}"
          assert_template_result "17", "{{ '17' | abs }}"
          assert_template_result "17", "{{ '-17' | abs }}"
          assert_template_result "0", "{{ 0 | abs }}"
          assert_template_result "0", "{{ '0' | abs }}"
          assert_template_result "17.42", "{{ 17.42 | abs }}"
          assert_template_result "17.42", "{{ -17.42 | abs }}"
          assert_template_result "17.42", "{{ '17.42' | abs }}"
          assert_template_result "17.42", "{{ '-17.42' | abs }}"
        end
    */
    @Test
    public void applyTest() throws RecognitionException {

        String[][] tests = {
                {"{{ 17 | abs }}", "17"},
                {"{{ -17 | abs }}", "17"},
                {"{{ '17' | abs }}", "17"},
                {"{{ '-17' | abs }}", "17"},
                {"{{ 0 | abs }}", "0"},
                {"{{ '0' | abs }}", "0"},
                {"{{ 17.42 | abs }}", "17.42"},
                {"{{ -17.42 | abs }}", "17.42"},
                {"{{ '17.42' | abs }}", "17.42"},
                {"{{ '-17.42' | abs }}", "17.42"}
        };

        for (String[] test : tests) {

            Template template = LiquidParser.newInstance().parse(test[0]);
            String rendered = String.valueOf(template.render());

            assertThat(rendered, is(test[1]));
        }
    }
}

File: liqp-core/src/test/kotlin/liqp/filter/AbsTest.kt
package liqp.filter

import assertk.assert
import assertk.assertions.isEqualTo
import liqp.LiquidParser
import liqp.createTestParser
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized
import org.junit.runners.Parameterized.Parameters

@RunWith(Parameterized::class)
class AbsTest(val template: String, val expected: String) {

  @Test
  fun applyTest() {
    val template = createTestParser{}.parse(template)
    val rendered = template.render()
    assert(rendered).isEqualTo(expected)
  }

  companion object {
    @JvmStatic @Parameters(name = "{0} = {1}")
    fun tests() = arrayOf(
        arrayOf("{{ 17 | abs }}", "17"),
        arrayOf("{{ -17 | abs }}", "17"),
        arrayOf("{{ '17' | abs }}", "17"),
        arrayOf("{{ '-17' | abs }}", "17"),
        arrayOf("{{ 0 | abs }}", "0"),
        arrayOf("{{ '0' | abs }}", "0"),
        arrayOf("{{ 17.42 | abs }}", "17.42"),
        arrayOf("{{ -17.42 | abs }}", "17.42"),
        arrayOf("{{ '17.42' | abs }}", "17.42"),
        arrayOf("{{ '-17.42' | abs }}", "17.42"))
  }
}
